crates/astria-core/src/sequencer/v1alpha1/block/mod.rs

```diff
@@ -667,35 +675,43 @@ impl SequencerBlock {
                     fee_asset_id: _,
                 }) = action
                 {
-                    let elem = rollup_base_transactions.entry(rollup_id).or_insert(vec![]);
+                    let elem = rollup_datas.entry(rollup_id).or_insert(vec![]);
+                    let data = RollupData::SequencedData(data).into_raw().encode_to_vec();
                     elem.push(data);
                 }
             }
         }
-        rollup_base_transactions.sort_unstable_keys();
+        for (id, deposits) in deposits {
+            rollup_datas.entry(id).or_default().extend(
+                deposits
+                    .into_iter()
+                    .map(|deposit| RollupData::Deposit(deposit).into_raw().encode_to_vec()),
+            );
+        }
+        rollup_datas.sort_unstable_keys();
```

Comment:
Can you add a comment above this line to bring home why sorting is necessary? I should have done that in the original impl. It's too easy to overlook this. E.g:

```rust
// XXX: The rollup data must be sorted by its keys before constructing the merkle tree.
// Since its constructed from non-deterministically ordered sources there is otherwise no
// guarantee that the same data will give the root.
```